spec/models/course_user_spec.rb

```diff
@@ -235,6 +235,19 @@
       end
     end
 
+    describe '#last_obtained_achievement' do
```

Comment:
@weiqingtoh you should be testing all the new methods you added, not just this one.